src/handlers/interventions-processor-handler.ts

```diff
@@ -55,24 +55,34 @@ async function processSQSRecord(itemFailures: SQSBatchItemFailure[], record: SQS
   try {
     const recordBody: TxMAEvent = JSON.parse(record.body);
     validateEvent(recordBody);
-    if (isTimestampNotInFuture(recordBody)) {
-      const intervention = getInterventionName(recordBody);
-      logger.debug('identified event: ' + intervention);
-      const itemFromDB = await service.retrieveRecordsByUserId(recordBody.user.user_id);
-      if (itemFromDB?.isAccountDeleted === true) {
-        logger.warn(`${LOGS_PREFIX_SENSITIVE_INFO} user ${recordBody.user.user_id} account has been deleted.`);
-        logAndPublishMetric(MetricNames.ACCOUNT_IS_MARKED_AS_DELETED);
-      } else {
-        logger.debug('retrieved item from DB ' + JSON.stringify(itemFromDB));
-        const statusResult = accountStateEngine.applyEventTransition(intervention, itemFromDB);
-        logger.debug('processed requested event, sending update request to dynamo db');
-        await service.updateUserStatus(recordBody.user.user_id, statusResult);
-      }
-    } else {
+    const intervention = getInterventionName(recordBody);
+
+    if (intervention === EventsEnum.IPV_IDENTITY_ISSUED && recordBody.extensions?.levelOfConfidence !== 'P2') {
+      logger.warn(`Received interventions has low level of confidence: ${recordBody.extensions?.levelOfConfidence}`);
+      logAndPublishMetric(MetricNames.CONFIDENCE_LEVEL_TOO_LOW);
+      return;
+    }
+
+    if (isTimestampInFuture(recordBody)) {
       itemFailures.push({
         itemIdentifier: record.messageId,
       });
+      return;
     }
+
+    logger.debug('identified event: ' + intervention);
```

Comment:
would it better to move this line to the top
```suggestion
    logger.debug(`${LOGS_PREFIX_SENSITIVE_INFO} Intervention received.`, { intervention});
```